### ISSUE TITLE
New option to exclude non-local filesystems

### ISSUE THREAD

**Issue by thomas-merz** [xsuchy/rpmconf#39]
Think about the following scenario which we have actually here in my company:  
* There is a NFS filesystem mounted
* You are trying to use rpmconf
* But it takes a long time to parse through the NFS filesystem because it has a large number
of files in it (much more than one millon, size much larger than one terabyte)
Therefore I'm looking to add an option similar to xdev in find to rpmconf so that such
huge filesystems which are not necessary can be excluded. I propose something like "-x" like an abbreviation for "-xdev" or "-l" for "local-only" or something like that.

**Comment by xsuchy** [comment]
Hmm, this is strange. What you are actually executing?

`rpmconf -a` does not go through the filesystem. It just goes through `rpm -qac` which is very fast.

On the other hand `rpmconf -c` go through "/etc", "/var", "/usr". But this command looks for orphan files. I.e. config of package which has been removed (e.g., with `dnf remove`). In my experience, this is very rarely needed as it finds only a few files. Especially if you previously run `rpmconf -a`.

**Comment by thomas-merz** [comment]
I was using your **very old version 0.3.4 from EPEL7** and there I found this find-line which "scans" some directories /etc /var /usr, where RPMs might change/update/copy some new config.

```
$ grep find $(which rpmconf)
        find /etc /var /usr -name '*.rpmnew' -o -name '*.rpmsave' | \
$ rpm -qa|grep rpmconf
rpmconf-0.3.4-1.el7.noarch
```

But underndeath /var we have a really huge NFS-share mounted (this is our pulp-mirror-server with maaaaaaany RPM files):
```
$ df -hit nfs
Filesystem                                                         Inodes IUsed IFree IUse% Mounted on
nfs-server:/MIRROR_PROD/var/lib/pulp    31M  1.2M   30M    4% /var/lib/pulp
$ $ df -ht nfs
Filesystem                                                          Size  Used Avail Use% Mounted on
nfs-server:/MIRROR_PROD/var/lib/pulp  1.5T  1.3T  280G  82% /var/lib/pulp
```

So the find command didn't ever finish successfully with "-c -d":
```
Orphaned .rpmnew and .rpmsave files:
/etc/nscd.conf.rpmsave
/var/lib/logrotate.status.rpmsave
/usr/sbin/rpmconf: line 215: read: read error: 1: Bad file descriptor
```

After I patched rpmconf by myself , it finished successfully in some seconds:
```
$ grep find $(which rpmconf)
        find /etc /var /usr -xdev -name '*.rpmnew' -o -name '*.rpmsave' | \
```

Or is there really the need to scan *all* directories in /var for rpmnew, rpmsave andsoon?
If this might be the case; i suggest an option someone can use on his/her own "risk" to ignore "other filesystems" (which usually are NFS mounted shares?!).

If "rpmconf -a -d" is the solution/option for me to use to only show rpmnew and rpmsave then I missunderstood and even didn't try "-a" option :-( Then you have also helped me a lot 👍  So I would only miss orphaned config files with "-a", am I right?

**Comment by thomas-merz** [comment]
Option "-a" is "interactive"-only and there no option to "show me only" in combination with "-a" available?!

So if I use "-c -d" then I will get a fine list to examine later and use rpmconf in scripts; but it fails with huge "other filesystems" submounted 👎 
So if I use "-a" it doesn't fail with huge "other filesystems" submounted; but I can't use this in my scripts and have to do it interactively 👎 

Can you solve this unfortunate situation? 😄

**Comment by xsuchy** [comment]
> So I would only miss orphaned config files with "-a", am I right?

Right.

> Can you solve this unfortunate situation?

Ok. So we have two issues in this one:

1 ) the new option to exclude a path - this is still valid, albeit not an as big priority with 1.x version which normally traverses very fast using -a. I will track it under this issue.

2) non-interactive mode for -a. I will track it as https://github.com/xsuchy/rpmconf/issues/41

**Comment by xsuchy** [comment]
Implemented in 103dd33
